src/Facade/Services/StoresFunctionCodeService.cs

```diff
@@ -38,12 +41,23 @@ protected override void DeleteOrObsoleteResource(StoresFunction entity, IEnumera
             throw new NotImplementedException();
         }
 
-        protected override Expression<Func<StoresFunction, bool>> FilterExpression(StoresFunctionResource searchResource)
+        protected override Expression<Func<StoresFunction, bool>> FilterExpression(StoresFunctionRequestResource searchResource,
+            IEnumerable<string> privileges = null)
         {
+            if (searchResource.OnlyAllowed == true)
+            {
+                if (privileges.Any())
+                {
+                    var allowedFunctions = privileges.Where(s => s.StartsWith("stores.requisitions.")) 
```

Comment:
should we change the convention to something like stores.requisitions.functions.XXXX now to stop us running into  problems with book, amend etc etc